lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3329,7 +3329,17 @@ private function isIdentifierEquals($entity1, $entity2)
             ? $this->entityIdentifiers[$oid2]
             : $this->identifierFlattener->flattenIdentifier($class, $class->getIdentifierValues($entity2));
 
-        return $id1 === $id2 || implode(' ', $id1) === implode(' ', $id2);
+        return $id1 === $id2 || $this->hashIdentifier($id1) === $this->hashIdentifier($id2);
+    }
+
+    /**
+     * @param array $identifier
+     *
+     * @return string
+     */
+    private function hashIdentifier(array $identifier)
+    {
+        return serialize(array_values(array_map('strval', $identifier)));
```

Comment:
I agree that it looks like pennies on this side, but we should still care very much for anything that is `O(n)` or in the same league.

We can't do miracles, and 2% here is indeed not bad, but we should strive for better performance also to get rid of the silly/misplaced "ORMs are slow" reputation.

We did a lot of research on this for 3.x, so we will see some impact there (better performance), but we should never just go full "oh, nobody cares".